mpas_analysis/ocean/plot_transect_subtask.py

```diff
@@ -0,0 +1,407 @@
+            the subtitle of the group in which this plot belongs (or blank
+            if none)
+
+        groupLink : str
+            a short name (with no spaces) for the link to the gallery group
+
+        galleryName : str
+            the name of the gallery in which this plot belongs
+
+        sectionName : str
+            the name of the section where the color map and range is defined
+
+        diffTitleLabel : str, optional
+            the title of the difference subplot
+
+        Authors
+        -------
+        Xylar Asay-Davis
+        """
+        self.outFileLabel = outFileLabel
+        self.fieldNameInTitle = fieldNameInTitle
+        self.mpasFieldName = mpasFieldName
+        self.refFieldName = refFieldName
+        self.refTitleLabel = refTitleLabel
+        self.diffTitleLabel = diffTitleLabel
+        self.unitsLabel = unitsLabel
+
+        # xml/html related variables
+        self.imageCaption = imageCaption
+        self.galleryGroup = galleryGroup
+        self.groupSubtitle = groupSubtitle
+        self.groupLink = groupLink
+        self.galleryName = galleryName
+
+        self.fieldNameInTitle = fieldNameInTitle
+        self.thumbnailDescription = self.season
+
+        self.configSectionName = configSectionName
+        # }}}
+
+    def setup_and_check(self):  # {{{
+        """
+        Perform steps to set up the analysis and check for errors in the setup.
+
+        Authors
+        -------
+        Xylar Asay-Davis
+        """
+        # first, call setup_and_check from the base class (AnalysisTask),
+        # which will perform some common setup, including storing:
+        #     self.runDirectory , self.historyDirectory, self.plotsDirectory,
+        #     self.namelist, self.runStreams, self.historyStreams,
+        #     self.calendar
+        super(PlotTransectSubtask, self).setup_and_check()
+
+        config = self.config
+        self.startYear = config.getint('climatology', 'startYear')
+        self.endYear = config.getint('climatology', 'endYear')
+        self.startDate = config.get('climatology', 'startDate')
+        self.endDate = config.get('climatology', 'endDate')
+
+        mainRunName = config.get('runs', 'mainRunName')
+
+        self.xmlFileNames = []
+
+        prefixPieces = [self.outFileLabel]
+        prefixPieces.append(self.transectName)
+        prefixPieces.append(mainRunName)
+        years = 'years{:04d}-{:04d}'.format(self.startYear, self.endYear)
+        prefixPieces.extend([self.season, years])
+
+        self.filePrefix = '_'.join(prefixPieces)
+
+        self.xmlFileNames.append('{}/{}.xml'.format(self.plotsDirectory,
+                                                    self.filePrefix))
+        # }}}
+
+    def run_task(self):  # {{{
+        """
+        Plots a comparison of ACME/MPAS output to SST, MLD or SSS observations
+        or a reference run
+
+        Authors
+        -------
+        Luke Van Roekel, Xylar Asay-Davis, Milena Veneziani
```

Comment:
@gstreletz, remove Luke and Milena here (oops!) and add maybe add yourself here.